### PR TITLE
Fixes#80 - Source Field added

### LINKED ISSUE #80: Feature Request - extra fields
Hi Laurent, I'm not a programer (I use code I find on the web to create my PowerShell and Javascript queries!) I'd find it really useful to have a field for:
- Source - so I can find the original code web site
- Works (Yes/No) I've tried a large amount of code snippets that don't work in my environment - it would be  nice to be able to save an tag these, so I don't keep re-trying something that doesn't work.
This information could be saved as a comment in the snippet, but it would be nice to be able to filter and search on these criteria as well.
Thanks very much for your help.
Kind regards
Gerhard

### PR DESCRIPTION
**Pull request by kartik-budhiraja**
Fixes #80, extra field added for adding a source to the notes.
The source shows up as an icon that takes to the link on click.

### CODE REVIEW

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.vue [hunk 1]
```diff
@@ -65,6 +66,7 @@ export default {
           this.note.name = `${noteType}:${generateNoteName()}`;
         }
 
+console.log(this.note)
```
Comment (lauthieb) [inline src/renderer/components/modals/create-note-modal/CreateNoteModal.vue]:
```suggestion
```

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.html [hunk 2]
```diff
@@ -24,6 +24,15 @@ <h2 class="text-red" v-if="displayDupError">Oh sorry, you can't have duplicated
         </b-input>
       </b-field>
 
+      <b-field horizontal label="Source">
```
Comment (lauthieb) [inline src/renderer/components/modals/create-note-modal/CreateNoteModal.html]: Maybe this `v-model` could be renamed to `note.sourceUrl`, what do you think?

**Comment by lauthieb** [comment]
Where is the icon?
I don't see it.
image

**Comment by lauthieb** [comment]
@kartik-budhiraja Can you please fix conflicts and update your branch with just your modifications?
Sorry for the inconvenience it was mandatory to force push master after several tries to fix this important bug #110

Here you can download the new version 1.2.3 : https://github.com/lauthieb/code-notes/releases/tag/1.2.3

Thanks a lot in advance for this.

**Comment by kartik-budhiraja** [comment]
@lauthieb Fixed the branch!!

**Comment by kartik-budhiraja** [comment]
Screen Shot 2019-12-08 at 3 48 39 PM

Screenshot with the source icon on the top

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.vue (hunk 1 above)
Comment (kartik-budhiraja) [reply src/renderer/components/modals/create-note-modal/CreateNoteModal.vue]: Done!

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.html (hunk 2 above)
Comment (kartik-budhiraja) [reply src/renderer/components/modals/create-note-modal/CreateNoteModal.html]: Done!

**Comment by lauthieb** [comment]
@kartik-budhiraja thanks ! 
on the top right corner?

**Comment by kartik-budhiraja** [comment]
@lauthieb yes, right next to the edit icon.

File: src/renderer/components/modals/create-note-modal/CreateNoteModal.html [hunk 3]
```diff
@@ -24,6 +24,15 @@ <h2 class="text-red" v-if="displayDupError">Oh sorry, you can't have duplicated
...
+        <b-input
+          type="url"
+          ref="noteSource"
+          v-model="note.sourceUrl"
+          placeholder="Source">
```
Comment (lauthieb) [inline src/renderer/components/modals/create-note-modal/CreateNoteModal.html]:
```suggestion
          placeholder="Source URL">
```

File: src/renderer/components/modals/update-note-modal/UpdateNoteModal.html [hunk 4]
```diff
@@ -25,6 +25,15 @@ <h2 class="text-red" v-if="displayDupError">Oh sorry, you can't have duplicated
         </b-input>
       </b-field>
 
+      <b-field horizontal label="Source">
+        <b-input
+          type="url"
+          ref="noteSource"
+          v-model="note.sourceUrl"
+          placeholder="Source">
```
Comment (lauthieb) [inline src/renderer/components/modals/update-note-modal/UpdateNoteModal.html]:
```suggestion
          placeholder="Source URL">
```

**Comment by lauthieb** [comment]
Hi :)
Can you please apply my suggestions?

**Comment by kartik-budhiraja** [comment]
Done :)

**Comment by lauthieb** [comment]
Hi @kartik-budhiraja,
I've just marked your Pull Request as closed because it has not had recent activity. 
It will be closed if no further activity occurs. 
Feel free to reopen it if you have time to contribute and fix conflicts... I haven't got any time to help you on this sorry...
Thanks again for your contributions.

**Comment by lauthieb** [comment]
- [ ]

**Pull request closed without merging**